director/monitor_test.go

```diff
@@ -0,0 +1,395 @@
+				if tc.expectTestToStart {
+					assert.True(t, started, tc.description)
+				} else {
+					assert.False(t, started, tc.description)
+				}
+			case <-time.After(time.Millisecond * 100):
+				if tc.expectTestToStart {
+					t.Errorf("Test case %s: expected test to start but timed out", tc.name)
+				}
+			}
+
+			// For cache expiration test, wait for the ad to expire
+			if tc.expectTestToStop {
+				// Wait for TTL to expire plus a bit more
+				time.Sleep(tc.cacheTTL + time.Millisecond*150)
+
+				// Check if test stopped due to cache expiration
+				select {
+				case <-testStopped:
+					// Test stopped as expected when cache expired
+				case <-time.After(time.Millisecond * 200):
+					t.Errorf("Test case %s: expected test to stop when cache expired but it didn't", tc.name)
+				}
+			}
+
+			// Wait for context cancellation to clean up
+			<-ctx.Done()
+		})
+	}
+}
+
+// TestDirectorTestDowntimeLogic specifically tests the downtime checking logic
+func TestDirectorTestDowntimeLogic(t *testing.T) {
+	now := time.Now().UTC().UnixMilli()
+
+	testCases := []struct {
+		name              string
+		downtime          server_structs.Downtime
+		expectSkip        bool
+		description       string
+	}{
+		{
+			name: "active-downtime",
+			downtime: server_structs.Downtime{
+				StartTime: now - 1000,
+				EndTime:   now + 1000,
+			},
+			expectSkip:  true,
+			description: "Should skip when downtime is active",
+		},
+		{
+			name: "past-downtime",
+			downtime: server_structs.Downtime{
+				StartTime: now - 2000,
+				EndTime:   now - 1000,
+			},
+			expectSkip:  false,
+			description: "Should not skip when downtime has ended",
+		},
+		{
+			name: "future-downtime",
+			downtime: server_structs.Downtime{
+				StartTime: now + 1000,
+				EndTime:   now + 2000,
+			},
+			expectSkip:  false,
+			description: "Should not skip when downtime hasn't started",
+		},
+		{
+			name: "indefinite-active-downtime",
+			downtime: server_structs.Downtime{
+				StartTime: now - 1000,
+				EndTime:   server_structs.IndefiniteEndTime,
+			},
+			expectSkip:  true,
+			description: "Should skip when in indefinite downtime",
+		},
+		{
+			name: "downtime-start-equals-now",
+			downtime: server_structs.Downtime{
+				StartTime: now,
+				EndTime:   now + 1000,
+			},
+			expectSkip:  true,
+			description: "Should skip when downtime starts exactly now",
+		},
+		{
+			name: "downtime-end-equals-now",
+			downtime: server_structs.Downtime{
+				StartTime: now - 1000,
+				EndTime:   now,
+			},
+			expectSkip:  true,
+			description: "Should skip when downtime ends exactly now",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			currentTime := now
+			hasActiveDowntime := false
+
+			// This is the exact logic from LaunchPeriodicDirectorTest
+			if tc.downtime.StartTime <= currentTime && (tc.downtime.EndTime >= currentTime || tc.downtime.EndTime == server_structs.IndefiniteEndTime) {
+				hasActiveDowntime = true
+			}
```

Comment:
It's bad form to duplicate the logic for a test -- instead, please make this a helper function that can be called in both the source code and the test to guarantee the test runs against the real code, even if it's edited in the future.